db/migrations/20230811005835_yen_gain_update.php

```diff
@@ -0,0 +1,121 @@
+<?php
+declare(strict_types=1);
+
+use Phinx\Migration\AbstractMigration;
+
+final class YenGainUpdate extends AbstractMigration
+{
+    /**
+     * Change Method.
+     *
+     * Write your reversible migrations using this method.
+     *
+     * More information on writing migrations is available here:
+     * https://book.cakephp.org/phinx/0/en/migrations.html#the-change-method
+     *
+     * Remember to call "create()" or "update()" and NOT "save()" when working
+     * with the Table class.
+     */
+    public function up(): void
+    {
+        // Locked out accounts
+       $this->execute("ALTER TABLE `users`
+            ADD COLUMN `login_attempt_time` INT NOT NULL DEFAULT 0 AFTER `failed_logins`,
+            ADD COLUMN `last_malicious_ip` TEXT AFTER `login_attempt_time`
+        ");
+        // Chat deletion
+        $this->execute("ALTER TABLE `chat` ADD COLUMN `deleted` INT(1) NOT NULL DEFAULT 0");
+        // NPC stuff
+        $this->execute("ALTER TABLE `ai_opponents` CHANGE `money` `money_multiplier` SMALLINT(3) NOT NULL;");
+        // Rank 1 AI
+        $this->execute("UPDATE `ai_opponents` SET `money_multiplier`=1 WHERE `ai_id`=1");
+        $this->execute("UPDATE `ai_opponents` SET `money_multiplier`=2 WHERE `ai_id`=2");
+        $this->execute("UPDATE `ai_opponents` SET `money_multiplier`=2 WHERE `ai_id`=3");
```

Comment:
Hmm, what's the intent of this? This strikes me as something we could statically calculate

Reply:
This is intended to take the place of the static money gains that are currently set for each AI. Example: Instead of saying Academy Bully gives 210 yen, it gives 1 multiplier.

We could totally remove this and allow arena logic and remove set rates in the NPC DB.